### ISSUE TITLE
Follow up questions feature does not seem to be working

### ISSUE THREAD

**Issue by kcordoc** [Azure-Samples/azure-search-openai-demo#240]
> Please provide us with the following information:
> ---------------------------------------------------------------

### This issue is for a: (mark with an `x`)
```
- [X] bug report -> please search issues before submitting
- [ ] feature request
- [ ] documentation issue or request
- [ ] regression (a behavior that used to work and stopped in a new release)
```

### Minimal steps to reproduce
> run the app on local server.

### Any log messages given by the failure
>none

### Expected/desired behavior
> after completion, you should be able to see follow up questions when follow-up questions are enabled in 'developer settings'

### OS and Version?
> Windows 7, 8 or 10. Linux (which distribution). macOS (Yosemite? El Capitan? Sierra?)
macOS

### azd version?
> run `azd version` and copy paste here.

### Versions
>1.0.0

### Mention any other details that might be useful 
I have deployed my app and the feature does not work

> ---------------------------------------------------------------
> Thanks! We'll be in touch soon.

**Comment by ArunEPRO** [comment]
Same for me as well. The follow up questions did not work.

**Comment by anderl80** [comment]
Same over here. Need to enable in developer settings but nothing happens.

**Comment by jomieljaniuk** [comment]
+1
It seems like Completitions ignore instruction to generate follow up questions in our prompt and generate only answer for current question.

**Comment by WhoIsDoctorWho** [comment]
Are there any update on this?

**Comment by sreesankar12** [comment]
I also encountered the issue. Could someone provide the correct prompt to receive the follow up question accurately?

**Comment by pamelafox** [comment]
The follow-up questions feature now works, but ChatGPT does not format the questions in a reliable manner. Sometimes it puts them inside angle brackets as requested, but it often puts them in a numbered list. That means the UI is not always able to render them as clickable links.

I've discussed follow-up questions with colleagues working on similar codebases, and we've identified three approaches to receiving follow-up questions in a well structured way so that they can become links in the UI:

* Post-processing the response to look for the variations in how ChatGPT responds. That's fairly brittle however. (Approach used by [the Javascript sample](https://github.com/Azure-Samples/azure-search-openai-javascript/blob/67dac3efd39145cc0c28835d41c5c39e1e48b428/packages/chat-component/src/utils/index.ts#L47)) 
* Using the "function call" feature of ChatGPT to receive a list of follow-up questions in the response.  (Which I just prototyped for this repo [in a branch](https://github.com/Azure-Samples/azure-search-openai-demo/compare/main...pamelafox:azure-search-openai-demo:followup?expand=1))
* Make an additional call to ChatGPT to specifically ask it to generate follow-up questions. (Approach used by [the C# sample](https://github.com/Azure-Samples/azure-search-openai-demo-csharp/blob/b14d971be1a9305ad9c76351967138d9f132ef74/app/backend/Services/ReadRetrieveReadChatService.cs#L33))

Each approach adds complexity in different parts of the codebase, so we haven't determined which way is the best.
